specs/protocol/holocene/configurability.md

```diff
@@ -0,0 +1,89 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Configurability](#configurability)
+  - [Overview](#overview)
+  - [Constants](#constants)
+    - [`ConfigType`](#configtype)
+  - [`SystemConfig`](#systemconfig)
+    - [Interface](#interface)
+  - [`OptimismPortal`](#optimismportal)
+    - [Interface](#interface-1)
+      - [`setConfig`](#setconfig)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Configurability
+
+## Overview
+
+The `SystemConfig` and `OptimismPortal` are updated with a new flow for chain
+configurability. This new flow reduces the need for the `ConfigUpdate` events
+emitted by the `SystemConfig`, which adds great cost to the fault proof program's
+execution as it results in the program to iterate through all possible logs to
+ensure that it has observed all of the events. It also enables the L1 Attributes
+transaction to not post duplicate data between transactions which reduces database
+size growth over time.
+
+## Constants
+
+### `ConfigType`
+
+The `ConfigType` enum represents configuration that can be modified.
+
+| Name | Value | Description |
```

Comment:
Now that I look at this, this is defining the enum value, that is the key in `setConfig`, we should also define the value in terms of the `bytes` in `setConfig(ConfigType,bytes)`. This is useful because the serialization is different for each type

Reply:
Currently, all the `StaticConfig.encodeXXX` functions are implemented as wrappers around `abi.encode`. Should we consider establishing this as a standard practice?

Reply:
That library is specific for encoding and decoding the values passed through the `setConfig` API. We don't necessarily need to ABI encode them, we could use `calldataload` directly, but its most simple to just `abi.encode`